ldap/servers/slapd/pw.c

```diff
@@ -1903,7 +1906,7 @@ pw_is_pwp_admin(Slapi_PBlock *pb, passwdPolicy *pwp)
     slapi_pblock_get(pb, SLAPI_REQUESTOR_ISROOT, &is_requestor_root);
 
     /* first check if it's root */
-    if (is_requestor_root) {
+    if (is_requestor_root && !ignore_rootdn) {
```

Comment:
It could also be something like 
if (is_requestor_root && (ignore_rootdn == PWP_ADMIN_OR_ROOTDN)) {